lib/middleware/http-proxy.js

```diff
@@ -1,18 +1,20 @@
 const log = require('@ui5/logger').getLogger('ui5-middleware-http-proxy')
 const request = require('request')
 const dotenv = require('dotenv')
+const lodash = require('lodash')
 
 // load .env
 dotenv.config()
 
-const _setAuthEnvironment = auth => {
+const _getAuth = ({ auth = {} }) => {
   const { HTTP_PROXY_AUTH_USER, HTTP_PROXY_AUTH_PASS } = process.env
-  if (!auth.user) {
+  if (!auth.user && HTTP_PROXY_AUTH_USER) {
     auth.user = HTTP_PROXY_AUTH_USER
   }
-  if (!auth.pass) {
+  if (!auth.pass && HTTP_PROXY_AUTH_PASS) {
     auth.pass = HTTP_PROXY_AUTH_PASS
   }
+  return !lodash.isEmpty(auth) ? auth : null
```

Comment:
I am not a fan of big dependencies for simple checks. Wouldn't `return Object.keys(auth).length ? auth : null` be enough?